wire-library/wire-moshi-adapter/src/main/java/com/squareup/wire/MessageJsonAdapter.kt

```diff
@@ -65,14 +69,30 @@ internal class MessageJsonAdapter<M : Message<M, B>, B : Message.Builder<M, B>>(
       fieldType = Types.newParameterizedType(List::class.java, fieldType)
     }
 
-    val syntheticQualifier: Class<out Annotation>? = when {
-      fieldBinding.singleAdapter() === ProtoAdapter.UINT64 -> Uint64::class.java
-      fieldBinding.label == WireField.Label.OMIT_IDENTITY -> OmitIdentity::class.java
-      else -> null
+    val isProto2 = defaultAdapter.syntax == Syntax.PROTO_2
+    val isProto3 = defaultAdapter.syntax == Syntax.PROTO_3
+    val syntheticQualifiers = mutableSetOf<Class<out Annotation>>()
```

Comment:
wanna change this from a `val MutableSet` to a `var Set`? You’ll get better performance!